src/tree/mod.rs

```diff
@@ -605,37 +625,37 @@
         // Note that zip makes it so only the first free_leaves().len() nodes are taken.
         let free_leaves = self.free_leaves();
         let free_leaves_len = free_leaves.len();
-        for (new_kp, leaf_index) in new_kps.iter().zip(free_leaves) {
-            self.nodes[leaf_index] = Node::new_leaf(Some((*new_kp).clone()));
+        for (&new_kp, leaf_index) in new_kps.iter().zip(free_leaves) {
+            self.nodes[leaf_index] = Node::new_leaf(Some((new_kp).clone()));
```

Comment:
Same here with the `clone`.

Reply:
Same as above, without a refactor of the proposal handling, I believe that we have to clone somewhere.